### PR TITLE
[BUGFIX] manage spawned processes

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##
Terminate spawned processes if SIGTERM is received. 

TODO: add unit test

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by mli** [comment]
Job PR-725/3 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-725/3/index.html

**Comment by mli** [comment]
Job PR-725/6 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-725/6/index.html

**Comment by szha** [comment]
Looks like the unittest jobs would hang after this change. This happens even for serial tests. The log shows segfault and an assertion error from pytest coverage collector. http://ci.mxnet.io/blue/organizations/jenkins/GluonNLP-py3-cpu-unittest/detail/PR-725/7/pipeline/80#step-162-log-1712

**Comment by mli** [comment]
Job PR-725/7 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-725/7/index.html

**Comment by mli** [comment]
Job PR-725/8 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-725/8/index.html

File: src/gluonnlp/data/stream.py [hunk 1]
```diff
@@ -384,10 +387,21 @@ def __iter__(self):
...
         else:
             return _ThreadPrefetcher(self._stream, self._num_prefetch,
                                      seed=seed, np_seed=np_seed,
                                      mx_seed=mx_seed)
+
+def _sigterm_handler(sig, _): # pylint: disable=unused-argument
+    """Handler for SIGTERM signal"""
+    for process in _managed_processes:
+        process.terminate()
+    sys.exit(0)
```
Comment (leezu) [inline src/gluonnlp/data/stream.py]: Should not exit with exit-code "successful" if exiting due to a signal.

**Pull request closed without merging**